prov/efa/src/efa_fabric.c

```diff
@@ -467,10 +467,23 @@ static int efa_alloc_fid_nic(struct fi_info *fi, struct efa_context *ctx,
 }
 
 #if HAVE_LIBCUDA
-static int efa_get_gdr_support(char *device_name)
+/*
+ * efa_get_gdr_support() check if GPUDirect RDMA is supported by
+ * reading from sysfs file "class/infiniband/<device_name>/gdr"
+ * and set content of gdr_support accordingly.
+ *
+ * Return value:
+ *   on success, return 0, and gdr_support is set to either 0 or 1.
+ *   on failure, a negative error code is returned.
+ *   Note this function will treat non-existent sysfs file as success
+ *   and set gdr_support to 0.
+ */
+static int efa_get_gdr_support(char *device_name, int *gdr_support)
```

Comment:
Why do we need to pass an int pointer? Why not just return <0 for fail, and 0/1 for success? Then we can avoid the assert, etc.

Reply:
change in new revision.